packages/react-components/tests/task-summary/task-summary-accordion.test.tsx

```diff
@@ -0,0 +1,129 @@
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import React from 'react';
+import { render } from '@testing-library/react';
+import { TaskSummaryAccordion, TaskSummaryAccordionInfo } from '../../lib';
+import { getActorFromStatus, sortTasksByState } from '../../lib/task-summary/task-summary-utils';
+
+const taskSummary = {
+  'af8faee9-84ca-41ea-8bb6-8493cc9f824c': {
+    end_time: { sec: 0, nanosec: 0 },
+    start_time: { sec: 0, nanosec: 0 },
+    state: 1,
+    status:
+      'Moving [tinyRobot/tinyRobot1]: ( 9.81228 -6.98942 -3.12904) -> ( 6.26403 -3.51569  1.16864) | Remaining phases: 1 | Remaining phases: 6',
+    submission_time: { sec: 0, nanosec: 0 },
+    task_id: 'af8faee9-84ca-41ea-8bb6-8493cc9f824c',
+  },
+  'am8faee9-84ca-41ea-8bb6-8493cc9f8249': {
+    end_time: { sec: 0, nanosec: 0 },
+    start_time: { sec: 0, nanosec: 0 },
+    state: 0,
+    status:
+      'Moving [tinyRobot/tinyRobot2]: ( 9.81228 -6.98942 -3.12904) -> ( 6.26403 -3.51569  1.16864) | Remaining phases: 1 | Remaining phases: 6',
+    submission_time: { sec: 0, nanosec: 0 },
+    task_id: 'am8faee9-84ca-41ea-8bb6-8493cc9f8249',
+  },
+};
+
+describe('Renders correctly', () => {
+  let task: RomiCore.TaskSummary;
+  beforeEach(() => {
+    task = Object.assign({}, Object.values(taskSummary)[0]);
+  });
+
+  test('Renders tree items', () => {
+    const tasks = Object.values(taskSummary);
+    const root = render(<TaskSummaryAccordion tasks={tasks} />);
+    tasks.forEach((task) => {
+      expect(root.getByText(task.task_id).textContent).toBe(task.task_id);
+    });
+  });
+
+  test('Show description below the id if the task has an actor', () => {
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    const actor = getActorFromStatus(task.status);
+    if (!actor) throw new Error('An actor is required to run this test');
+    const classes = root.getByText(actor[0]).className;
+    expect(classes).toContain('makeStyles-taskActor');
+  });
+
+  test('Does not show description below the id if the task has no actor', () => {
+    task.status = 'Finished';
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.container.querySelector('[id=task-actor]')).toBeFalsy();
+  });
+});
+
+describe('Components gets the correct style on specifics states', () => {
+  let task: RomiCore.TaskSummary;
+  beforeEach(() => {
+    task = Object.assign({}, Object.values(taskSummary)[0]);
+  });
+
+  test('active style is applied ', () => {
+    task.state = RomiCore.TaskSummary.STATE_ACTIVE;
+    const root = render(<TaskSummaryAccordion tasks={[task]} />);
+    expect(root.getByText(task.task_id).parentElement?.className).toContain('makeStyles-active');
```

Comment:
I'm not sure which practice to follow. I looked at material-ui's test and there are a lot of tests on the css class like this. But on the other hand RTL does not recommend testing class names because that is not what an user cares about.

Which philosophy do you think makes more sense?

Reply:
Yeah, I think in our case both will help, but we should establish priority. I think we should go with the RTL way as a top priority for all our components. I'm imagining that the RTL way will help a lot with the code refactor, because since we don't have to look for a specific class or id, if we change the id or the component in the refactor and the component looks the same we will not have to maintain (update) that test.

For the material-ui components, If there are components associated with `Material-ui` and getting the values from those components became too convoluted (e.g.: check the background's color in a component with a lot of parents htmlElement.parent().parent().parent().leftchild()....), I think we should go with the material-ui way and check for a css class.

The problem with using both philosophies as I proposed is where to draw a line to say that something is "convoluted" and worth using the css class verification.